app/js/components/queryBuilder/queryParamSelector/queryParamSelector.component.js

```diff
@@ -55,7 +59,28 @@ const QueryParamSelectorComponent = {
             }
 
 
+            function getSearchResults(searchQuery) {
+              vm.isSearching = true;
+
+              // pass along the user input query and selected mutations list 
+              // to the appropriate service 
+              searchServices[vm.currentState()]
+                .query(searchQuery, vm.mutationsSet)
+                .then(response=>{
 
+                  $scope.$apply(()=>{
+                      if(response.length){
+                        vm.searchResults = _filteredSearchResults(response);
+                        vm.progressBar.goTo(`Add ${progressStateName}`);
+                      } else{
+                        vm.progressBar.goTo(`Search ${progressStateName}`);  
+                      }
+                      
+                      vm.isSearching = false;
+                  });
+                  
+                });//END searchServices[vm.currentState()]
```

Comment:
What if the query fails? Do you need a function to handle the error or it has been handled somewhere else?
By the way, the line may look nicer if you change it into:
```}); // END searchServices[vm.currentState()]```